packages/@sanity/block-tools/src/HtmlDeserializer/rules/html.ts

```diff
@@ -188,15 +195,19 @@ export default function createHTMLRules(
       },
     }, // Deal with list items
     {
-      deserialize(el, next) {
+      deserialize(el, next, block) {
         const tag = tagName(el)
         const listItem = tag ? HTML_LIST_ITEM_TAGS[tag] : undefined
         const parentTag = tagName(el.parentNode) || ''
+        const enabledListItem = resolveListItem(parentTag, options.enabledListTypes)
         if (!listItem || !el.parentNode || !HTML_LIST_CONTAINER_TAGS[parentTag]) {
           return undefined
         }
```

Comment:
This can be moved down below the conditional, in order to not do unnecessary work:

```suggestion
        if (!listItem || !el.parentNode || !HTML_LIST_CONTAINER_TAGS[parentTag]) {
          return undefined
        }
        
        const enabledListItem = resolveListItem(parentTag, options.enabledListTypes)
```

Reply:
Yes, changed this as well!